### ISSUE TITLE
Custom attribute running once only

### ISSUE THREAD

**Issue by aadamsx** [aldeed/meteor-simple-schema#182]
my select radio only runs the Custom attribute once, while the others run every time I submit.

The first on is the radio, the second one is just a input. The latter, custom, runs every time I hit submit and displays an validation error fine.  The former, radio, runs only once, the first time -- before I save off to the DB, and displays a validation error.

If I save, by not checking otherNamesUsed (which by passes the validation), and come back in and check this otherNamesUsed, only the input custom runs and validates.  The radio custom never runs again.

```
    nameChanged: {
        type: String,
        optional: true,
        allowedValues: ["No", "Yes"],
        custom: function () {
          if (this.field('otherNamesUsed').value === "Yes" && (this.value === undefined)) {
            return "required";
          }
        }
    },
    otherName1First: {
        type: String,
        optional: true,
        custom: function () {
          if (this.field('otherNamesUsed').value === "Yes"
            && (this.value === null || this.value === "" || this.value === undefined)
          ) {
            return "required";
          }
        }
    }
```

Other names used

```
    otherNamesUsed: {
      type: String,
      optional: false,
      allowedValues: ["No", "Yes"]
    }
```

Radio control and input control

```
{{> afFieldInput name=nameChanged type="select-radio-inline" options="allowed" template="bootstrap3-horizontal"}}
{{> afFieldInput name=otherName1First placeholder=placeHolder template="bootstrap3-horizontal"}}
```

**Comment by aadamsx** [comment]
Yes, even after all the time, it's still an issue.  The basic problem is, for radio controls only, on insert the custom function in the schema runs fine, but on update, it never runs -- no matter what I put in the custom.  At first I thought maybe it doesn't run in update mode if somehow the custom conditions are not met, but even if I just return a 'required', it sill will not run.  

For all other types of controls custom runs as it should.

aldeed:autoform               5.4.0
aldeed:autoform-select2       2.0.2
aldeed:collection2            2.3.3
Meteor 1.1.0.3

**Comment by serkandurusoy** [comment]
Could you try `!this.isSet` instead of `(this.value === null || this.value === "" || this.value === undefined)`

**Comment by sptessier** [comment]
I'm using aldeed:simple-schema@1.5.3 with aldeed:collection2@2.9.0 and ReactJS. 

I have a custom function (key level) that returns "required" only if the value of another field is true. It works fine on insert but on update it doesn't work since the conditionnaly required key is not part of the $set operator and therefore the custom function is not fired at all.

It seems like our friend @aldeed is refactoring the validation quite a lot because I can see that the validation flow is quite different between doValidation2 and the current doValidation1 functions. 

I think the convertModifierToDoc approach is promising. I thought implementing something similar in my app but I didn't want to lose the ability to have the isInsert and isUpdate context in my custom validation function. 

+1 for having this handled internally within SimpleSchema validation logic.

**Comment by aldeed** [comment]
@sptessier I realize your last comment was a long time ago, but I just wanted to mention that I gave up on the `convertModifierToDoc` approach because there are a lot of edge cases where I couldn't figure out a good solution. If anyone can figure it out, I agree it would be a pretty slick way to support modifiers without people who use the package needing to be aware of the modifier itself. One idea might be to skip validation for tricky modifiers like `$inc` or `$rename` and you'd have to opt-in to use them otherwise it would just throw an error when you try to use them.

**Comment by aldeed** [comment]
The Meteor SimpleSchema package is no longer maintained other than critical fixes to keep it running with each latest Meteor release. Closing non-critical issues in this repo. Anyone who cares about this may do one or more of the following:

- Switch to the [NPM package](https://github.com/aldeed/simple-schema-js). Be sure to adjust for the [breaking changes](https://github.com/aldeed/meteor-simple-schema/blob/master/CHANGELOG.md#200). There can be a lot of work involved in switching, but it is worth the effort.
- If this issue still occurs in the NPM package, [submit an issue there](https://github.com/aldeed/simple-schema-js/issues). Be sure to include a link to a reproduction repo or a PR that adds a failing test for the issue.
- If you have more time than money, submit a pull request to resolve any [issue labeled "help wanted"](https://github.com/aldeed/simple-schema-js/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) in the NPM package repo.
- If you have more money than time, [donate](https://opencollective.com/simple-schema-js/donate) to help support development if you are able
